torch/csrc/jit/codegen/cuda/python_frontend/python_bindings.cpp

```diff
@@ -187,46 +192,76 @@ void initNvFuserPythonBindings(PyObject* module) {
           py::arg("ndims"),
           py::arg("dtype") = torch::jit::fuser::cuda::DataType::Float,
           py::return_value_policy::reference)
-      .def_static(
+      .def(
           "define_tensor",
-          [](size_t ndims,
-             std::vector<bool> contiguity,
+          [](FusionDefinitionContextManager& self,
+             std::vector<int> sizes,
+             std::vector<int> strides,
              torch::jit::fuser::cuda::DataType dtype =
                  torch::jit::fuser::cuda::DataType::Float) -> TensorView* {
             TORCH_CHECK(
-                ndims == contiguity.size(),
-                "The number of dimensions specified does not match contiguity.");
-            return TensorViewBuilder()
-                .ndims(ndims)
-                .dtype(dtype)
-                .contiguity(std::move(contiguity))
-                .build();
+                sizes.size() == strides.size(),
+                "The number of sizes does not match the number of strides.",
+                sizes.size(),
+                strides.size());
+
+            std::vector<IterDomain*> domain_sizes;
+            for (const auto i : c10::irange(sizes.size())) {
+              if (sizes[i] == 1) {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(),
+                    self.fusionPtr()->oneVal(),
+                    ParallelType::Serial,
+                    IterType::BroadcastWithStride));
+              } else {
+                domain_sizes.push_back(IrBuilder::create<IterDomain>(
+                    self.fusionPtr()->zeroVal(), IrBuilder::create<Int>()));
+              }
+            }
+
+            std::vector<bool> contig_info(strides.size(), false);
+            for (int i = contig_info.size() - 1; i >= 0; --i) {
+              if (i == static_cast<int>(contig_info.size() - 1)) {
+                contig_info[i] = (strides[i] == 1);
+              } else {
+                contig_info[i] =
+                    (strides[i] == (strides[i - 1] * sizes[i - 1]));
```

Comment:
Caught you again here~

Per the line above, higher dimension should be the inner dimension, so here `i-1` should be `i+1` instead.

Reply:
Feel free to add test cases to the test in test_prims.py, too!

Reply:
Right now, the correct contiguity logic is causing a failure in this test that I am trying to track down.

```
            # Adds outermost, expands, and unsqueezes
            a = make_arg((1, 2, 3))
            b = make_arg((4, 1, 7, 2, 3, 3), low=0.0, high=0.0)
            result = fn(a, b, (1, 3, 4))

            self.assertEqual(result.shape, b.shape)
            a.unsqueeze_(3)
            a.unsqueeze_(1)
            a.unsqueeze_(0)
            self.assertEqual(a.expand_as(result), result)
```

Specifically the comparison of `a.expand_as(result)` is failing.